src/content_mapblock.cpp

```diff
@@ -172,6 +172,10 @@ static inline void getNeighborConnectingFace(v3s16 p, INodeDefManager *nodedef,
 		*neighbors |= v;
 }
 
+static int NeighbourToIndex(const v3s16 &pos) {
+	return (9 * pos.X + 3 * pos.Y + pos.Z) + 13;
```

Comment:
The bitshift method might be more intuitive, but much less efficient overall due to the index gaps causing more of a space requirement.  Multiplication by constant expressions are optimized by the compiler when it's possible to, so that argument is moot.  I'd rather not have something that burdensome instead because it "might possibly" aid with understanding.

I think a lot can be illustrated by demonstrating in comments where the form of this equation came from, e.g. "// This expression is a simplification of (3*3*1) * (pos.Z+1) + (3*1) * (pos.Y+1) + (1) * (pos.X+1)".